refactor(ecmascript): add `ToUint32` trait

sapphi-red opened the pull request:
Added `ToUint32` trait that corresponds to the [`ToUint32` abstract operation](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-touint32).

Boshen commented:
### Merge activity

* **Aug 24, 1:08 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
refactor(ecmascript): add `ToUint32` trait (#13272)

Added `ToUint32` trait that corresponds to the [`ToUint32` abstract operation](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-touint32).
